src/devtools/client/inspector/rules/rules.js

```diff
@@ -578,6 +578,11 @@ class RulesView {
    * to be refresh (e.g. when print media simulation is toggled).
    */
   async updateElementStyle() {
+    // Updating the style panels can take more than 100ms and blocks the browser from
```

Comment:
100ms! That's a pretty long time! What's so heavy about the code this function calls?

Reply:
Also, and I am no expert on how this works, but what if we instead did:

```
this.elementStyle.populate().then(() => this.updateRules())
```

without the `await`. Would that have a similar effect or am I missing something?

Reply:
No, we need the `setTimeout` in there because awaiting a resolved promise doesn't unblock the rendering. And (AFAIK) it doesn't make a difference if you use `await` or `.then()`.

Reply:
> 100ms! That's a pretty long time! What's so heavy about the code this function calls?

Yeah, that's a mystery to me too. I found and fixed (in this PR) 2 places that slowed the computation down a lot, but it's still slower than I would expect it to be. I'm pretty sure if I spent more time on this, I could find other places to speed up.